File: dfs/rangeSum.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: TreeNode, low: int, high: int) -> int:
        self.rangeSum = 0
        def dfs(root):
            if root:
                if low <= root.val <= high:
                    self.rangeSum += root.val
                if low < root.val:
                    dfs(root.left)
                if root.val < high:
                    dfs(root.right)
                    
        
        dfs(root)
        return self.rangeSum

File: dfs/rangeSum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int rangeSum = 0;
    public int rangeSumBST(TreeNode root, int low, int high) {
        dfs(root, low, high);
        return rangeSum;
    }
    public void dfs(TreeNode root, int low, int high) {
        if(root != null) {
            if(low <= root.val && root.val <= high) {
                rangeSum += root.val;
            }
            if(low < root.val) {
                dfs(root.left,low, high);
            }
            if( root.val < high) {
                dfs(root.right,low, high);
            }
        }
    }
}